resources/app/src/components/expert_panels/ReviewMembership.vue

```diff
@@ -20,152 +20,190 @@
         fields.value.push('coi_completed');
     }
 
-    const members = ref([]);
-    const chairs = computed(() => tableRows.value.filter(m => m.roles.includes('chair')));
-    const experts = computed(() => tableRows.value.filter(m => m.roles.includes('expert')));
-
-    const memberGroups = computed(() => [
-        {
-            title: 'Leadership',
-            members: chairs.value
-        },
-        {
-            title: 'Coordination',
-            members: tableRows.value.filter(m => m.roles.includes('coordinator'))
-        },
-        {
-            title: 'Biocuration',
-            members: tableRows.value.filter(m => m.roles.includes('biocurator'))
-        },
-        {
-            title: 'Expertise',
-            members: experts.value
-        }
-    ])
-
-    const delay = (ms = 1000) => new Promise((r) => setTimeout(r, ms));
-    const getPublications = async member => {
-        const baseUri = 'https://eutils.ncbi.nlm.nih.gov/entrez/eutils';
-        const searchUrl = `${baseUri}/esearch.fcgi?db=pubmed&term=${member.last_name},+${member.first_name}[author]&retmode=json&retmax=0`;
-
-        return axios.get(searchUrl)
-            .then(rsp => {
-                member.pubCount = rsp.data.esearchresult.count
-            })
-            .catch(async error => {
-                if (error.response.status == 429) {
-                    await delay();
-                    getPublications(member);
-                }
-            });
+   // const members = ref([]);
+
+
+   const members = ref([]);
+
+// Watch for changes in props.members and apply sorting and aggregation
+watch(() => props.members, async (to) => {
+    if (!to) {
+        return;
     }
 
-    watch(() => props.members, async to => {
-        if (!to) {
-            return;
+    // Copy the incoming members array
+    members.value = [...to];
+
+    // Sort members based on roles
+    members.value.sort((a, b) => {
+        if (a.roles.includes('chair') && !b.roles.includes('chair')) {
+            return -1;
+        }
+        if (!a.roles.includes('chair') && b.roles.includes('chair')) {
+            return 1;
+        }
+
+        if (a.roles.includes('expert') && !b.roles.includes('expert')) {
+            return -1;
         }
+        if (!a.roles.includes('expert') && b.roles.includes('expert')) {
+            return 1;
+        }
+
```

Comment:
Clearly only handles chart and expert for now. Has the overall order been specified for other roles?

Reply:
I see this later...